src/Cards/Card.java

```diff
@@ -0,0 +1,113 @@
+package Cards;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.IOException;
+import java.util.Objects;
+
+public class Card {
+
+    private String color;
+    private final Integer rank;
+    private Image cardFront;
+    public static String cardBackFilename = "card_back_blue";
+    private boolean faceUp;
+    public static String cardImageDirectory = "/Cards/cardImages", extension = ".png";
+    private String suit;
+
+    public Card(Suit suit, Integer rank) {
+        this.rank = rank;
+        switch (suit) {
+            case CLUBS -> {
+                this.suit = "clubs";
+                this.color = "black";
+            }
+            case DIAMONDS -> {
+                this.suit = "diamonds";
+                this.color = "red";
+            }
+            case HEARTS ->{
+                this.suit = "hearts";
+                this.color = "red";
+            }
+            case SPADES ->{
+                this.suit = "spades";
+                this.color = "black";
+            }
+        }
+        faceUp = false;
+
+        try {
+             System.out.println(cardImageDirectory + parsedCardFile(suit, rank));
+            ImageIcon imageIcon = new ImageIcon(getClass().getResource(cardImageDirectory +
+                    parsedCardFile(suit, rank)));
+            cardFront = imageIcon.getImage();
+        }catch(Exception e) {
+            System.err.println("Error: " + e.getMessage());
+        }
```

Comment:
[Flyweight](https://refactoring.guru/design-patterns/flyweight).

Does not fully apply here as cards do not actually repeat.

But let's assume you would duplicate cards, you needn't load the same image twice, but store them somewhere else (`CardImageMap -> Map<Tuple<Integer, Suit>, Image>`)